packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -142,11 +177,76 @@ export const treeNodeApi: TreeNodeApi = {
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
 	/**
-	 * Raised on a node right after a change is applied to one of its fields or the fields of a descendant node.
+	 * Raised on a node when one or more of its child nodes are replaced, i.e. when the properties of the node are
+	 * assigned to (this includes assigning `undefined` to a property to remove the child node there).
+	 *
+	 * @remarks
+	 * In particular, this event is not raised when:
+	 * - Properties of a child node change. Notably, updates to an array node or a map node (like adding or removing
+	 * elements/entries) will raise this event on the array/map node itself, but not on the node that contains the
+	 * array/map node as one of its properties.
+	 * - The node is moved to a different location in the tree or removed from the tree.
+	 * In this case the event is raised on the _parent_ node, not the node itself.
+	 *
+	 * Also note that value nodes do not have properties (child nodes), so this event is never raised for them.
+	 *
+	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
+	 * the client that made the original edit.
+	 * While a batch of edits will as a whole update the tree to the appropriate end state, no guarantees are made about
+	 * how many times this event will be raised during any intermediate states.
+	 * When it is raised, the tree is guaranteed to be in-schema.
+	 *
+	 * @privateRemarks
+	 * In terms of the internal implementation of tree, this fires when:
+	 * - The content of one or more of the node's fields changes (i.e., the field now contains a new node, or nothing if it
+	 * previously contained a node)
+	 * - For an array node, when the array is modified (i.e., an element is added, removed, or moved)
+	 * - For a map node, when a key is added, updated, or removed.
+	 *
+	 * This event occurs whenever the apparent contents of the node instance change, regardless of what caused the change.
+	 * For example, it will fire when the local client reassigns a child, when part of a remote edit is applied to the
+	 * node, or when the node has to be updated due to resolution of a merge conflict
+	 * (for example a previously applied local change might be undone, then reapplied differently or not at all).
+	 */
+	afterShallowChange(): void;
+
+	/**
+	 * Raised on a node when changes happen anywhere in the subtree (including itself) rooted at it.
```

Comment:
Not sure what the "(including itself)" is meant to convey here.

Reply:
Ah, I think I understand. I agree it's good to spell it out. My confusion stemmed from the fact that, with the current construction, "itself" refers to "the subtree" as opposed to "a node". I think "(including changes to the root of the subtree)" would be clearer, but I'm sure there are other good ways to phrase this.